hopsworks-web/yo/app/scripts/controllers/clusterUtilisationCtrl.js

```diff
@@ -55,11 +58,106 @@ angular.module('hopsWorksApp')
+                        success.data.forEach(function (status) {
+                          if (status.role === "datanode") {
+                            nbDataNodes++;
+                            if (status.status === "Started") {
+                              nbRunningDataNodes++;
+                            }
+                          }
+                          if (status.role === "namenode") {
+                            nbNameNodes++;
+                            if (status.status === "Started") {
+                              nbRunningNameNodes++;
+                            }
+                          }
+                        });
+                        if (nbRunningDataNodes > nbDataNodes * 2 / 3 && nbRunningNameNodes > nbNameNodes / 2) {
+                          self.HdfsStatus = 'status-success';
+                        } else if (nbRunningDataNodes > nbDataNodes * 1 / 3 && nbRunningNameNodes >= 1) {
+                          self.HdfsStatus = 'status-warning';
+                        } else {
+                          self.HdfsStatus = 'status-danger';
+                        }
+
+                      }, function (error) {
+                console.log("problem getting HDFS status");
+              });
+            };
+
+            var getYarnStatus = function () {
+              ClusterUtilService.getYarnStatus().then(
+                      function (success) {
+                        var nbNodeManagers = 0;
+                        var nbRunningNodeManagers = 0;
+                        var nbRMs = 0;
+                        var nbRunningRMs = 0;
+                        success.data.forEach(function (status) {
+                          if (status.role === "nodemanager") {
+                            nbNodeManagers++;
+                            if (status.status === "Started") {
+                              nbRunningNodeManagers++;
+                            }
+                          }
+                          if (status.role === "resourcemanager") {
+                            nbRMs++;
+                            if (status.status === "Started") {
+                              nbRunningRMs++;
+                            }
+                          }
+                        });
+                        if (nbRunningNodeManagers > nbNodeManagers * 2 / 3 && nbRunningRMs > nbRMs / 2) {
+                          self.YarnStatus = 'status-success';
+                        } else if (nbRunningNodeManagers > nbNodeManagers * 1 / 3 && nbRunningRMs >= 1) {
+                          self.YarnStatus = 'status-warning';
+                        } else {
+                          self.YarnStatus = 'status-danger';
+                        }
+
+                      }, function (error) {
+                console.log("problem getting HDFS status");
+              });
+            };
+
+            var getKafkaStatus = function () {
+              ClusterUtilService.getYarnStatus().then(
```

Comment:
@berthoug Kafka status uses getYarnStatus. Is this correct?

Reply:
Oups, nop, I'll correct that.